include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -144,7 +144,9 @@ MultigridPreconditionerBase<dim, Number>::initialize_levels(Triangulation<dim> c
   }
   else // h-MG is involved working on all mesh levels
   {
-    for(unsigned int h = 0; h < tria->n_global_levels(); h++)
+    for(unsigned int h = 0; h < (data.use_global_coarsening ? coarse_grid_triangulations.size() :
+                                                              tria->n_global_levels());
+        h++)
       h_levels.push_back(h);
```

Comment:
Just a tiny comment here: in such cases I would prefer brackets for readability (or a variable `n_h_levels`).

Reply:
Done in cd54f5c3 (force-pushed).